File: src/main/java/com/github/trcdeveloppers/clayium/gui/GuiHandler.java
package com.github.trcdeveloppers.clayium.gui;

import com.github.trcdeveloppers.clayium.blocks.machines.container.ClayWorktableContainer;
import com.github.trcdeveloppers.clayium.blocks.machines.tile.TileClayWorkTable;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;
import net.minecraftforge.fml.common.network.IGuiHandler;

import javax.annotation.Nullable;

public class GuiHandler implements IGuiHandler {
    public static final int ClayWorkTable = 1;

    @Nullable
    @Override
    public Object getServerGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        System.out.println("called guihandler");
        TileEntity te = world.getTileEntity(new BlockPos(x, y, z));
        if (te == null) { return null; }

        switch (ID) {
            case ClayWorkTable:
                System.out.println("server gui id 1");
                return new ClayWorktableContainer(player.inventory, (TileClayWorkTable) te);
        }
        return null;
    }

    @Nullable
    @Override
    public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        System.out.println("called guihandler");
        TileEntity te = world.getTileEntity(new BlockPos(x, y, z));
        if (te == null) { return null; }

        switch (ID) {
            case ClayWorkTable:
                System.out.println("client gui id 1");
                return new GuiClayWorkTable(player.inventory, (TileClayWorkTable) te);
        }
        return null;
    }
}

File: src/main/kotlin/com/github/trc/clayium/common/GuiHandler.kt
package com.github.trc.clayium.common

import com.github.trc.clayium.client.gui.GuiClayWorkTable
import com.github.trc.clayium.common.blocks.clayworktable.TileClayWorkTable
import com.github.trc.clayium.common.gui.ContainerClayWorkTable
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World
import net.minecraftforge.fml.common.network.IGuiHandler

// todo: move CWT to mUI
object GuiHandler : IGuiHandler {

    const val CLAY_WORK_TABLE = 1

    override fun getServerGuiElement(
        id: Int,
        player: EntityPlayer,
        world: World,
        x: Int,
        y: Int,
        z: Int
    ): Any? {
        val tile = world.getTileEntity(BlockPos(x, y, z)) ?: return null
        return when (id) {
            CLAY_WORK_TABLE -> ContainerClayWorkTable(player.inventory, tile as TileClayWorkTable)
            else -> null
        }
    }

    override fun getClientGuiElement(
        id: Int,
        player: EntityPlayer,
        world: World,
        x: Int,
        y: Int,
        z: Int
    ): Any? {
        val tile = world.getTileEntity(BlockPos(x, y, z)) ?: return null
        return when (id) {
            CLAY_WORK_TABLE -> GuiClayWorkTable(player.inventory, tile as TileClayWorkTable)
            else -> null
        }
    }
}
